### PR TITLE
Integration testing

### LINKED ISSUE #469: Create basic integration tests to ensure all pages of the app load successfully
Currently we have integration tests verifying that the gameboard page loads correctly. We should add very basic tests to ensure that each other pages loads successfully end to end as well.

### PR DESCRIPTION
**Pull request by camdenmoors**
Closes #469 
Adds tests for: 

- Application:
  - Navigation bar fields (Logged In + Logged Out)
  - Navigation bar fields (With Team + Without Team)
  - Game name shows on home page
  - Correct catchphrase is shown on home page
  - Game description is displayed on home page
  - Register button shows on home page when not logged in
  - Create/Join team shows up while logged in without a team
  - Join team shows up while logged in with a team (I think this should be changed to view team in a later commit)
  - Footer displays properly
 - Messages: 
   - Message header is visible
   - "No Messages" is show when there are no messages
   - Messages display contents correctly
 - Achievements: 
   - Achievement header shows
   - No achievements shows when there are no achievements
   - Achievement table displays correctly
 - Summary: 
   - Summary header shows
   - No teams shows when there are no teams
   - Summary table displays correctly
 - Team Pages:
   - Team creation:
      - Team creation header shows
      - Team creation form shows correctly
   - Join team:
      - Join team header shows
      - Join team table shows correctly
   - Team show/summary
      - Team show:
         - Team name + eligibility + points shows in header
         - Team user list table displays correctly
         - Team Prize Eligibility Status/Team Division Status displays correctly
      - Team summary:
          - Solve challenges table displays correctly

### CODE REVIEW

File: app/views/teams/show.html.haml [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
-    %th.span1= t('teams.team_users_table.team_leader_header')
-    %th= t('teams.team_users_table.email_header')
-    %th= t('teams.team_users_table.grade')
-    %th.span2= t('teams.team_users_table.prize_eligibility_header')
-    %th= t('teams.team_users_table.remove_user_header')
-    - if @team_captain
-      %th
-        = t('teams.team_users_table.change_captian_header')
+    %tr
+      %th.span1= t('teams.team_users_table.team_leader_header')
```
Comment (rbclark) [inline app/views/teams/show.html.haml:10]: It seems a little reduntant here for the parent key to be `teams` and then the child key to start with `team`, maybe rename this to `teams.user_table.team_leader_header`?

File: test/integration/application_display_modes_test.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      open_source_url: 'https://github.com/mitre-cyber-academy/ctf-scoreboard',
+      board_layout: 0
+    )
+  end
+
+  test 'navigation bar shows all fields when logged out' do
+    get "/"
+    
+    assert_select 'a[class=brand]', @game.organization, "Organization should show in navigation bar"
+    assert_select 'a[href=\/game]', I18n.t('application.navbar.challenges'), "Challenges should show in navigation bar"
```
Comment (rbclark) [inline test/integration/application_display_modes_test.rb:29]: Can you see if there is a way to access the route helpers here so we don't have to hardcode all of the `href` paths?

File: test/integration/application_display_modes_test.rb [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    assert_select 'a[href=\/game\/messages]', I18n.t('application.navbar.messages'), "Messages should show in navigation bar"
+    assert_select 'a[href=\/game\/achievements]', I18n.t('application.navbar.achievements'), "Achievements should show in navigation bar"
+    assert_select 'a[href=\/game\/summary]', I18n.t('application.navbar.summary'), "Summary should show in navigation bar"
+    assert_select 'a[href=https\:\/\/example\.com]', I18n.t('application.navbar.contact'), "Contact should show in navigation bar if contact_url is defined in game"
```
Comment (rbclark) [inline test/integration/application_display_modes_test.rb:33]: Can you try accessing this by calling `@game.contact_url`?

**Comment by rbclark** [comment]
Also, can you rebase this against master?

**Comment by camdenmoors** [comment]
moved to #516

**Pull request closed without merging**